fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/AgilityContextTest.java

```diff
@@ -414,6 +418,39 @@ void testAgilityContextAtomicAttribute(boolean useAgile) {
         }
     }
 
+    @Test
+    void testAgilityContextRecoveryPath() {
+        AtomicInteger refInt = new AtomicInteger(0);
+        try (FDBRecordContext context = openContext()) {
+            final AgilityContext agilityContext = getAgilityContext(context, true);
+            final Subspace subspace = path.toSubspace(context);
+            byte[] key = subspace.pack(Tuple.from(2023, 3));
+            byte[] val = Tuple.from(800, "Green eggs and ham", 0).pack();
+            agilityContext.set(key, val);
+            agilityContext.abortAndClose();
+            IntStream range = IntStream.rangeClosed(1, 10);
+            range.parallel().forEach(i -> {
```

Comment:
Is the goal of the parallelism here to make sure that `applyInRecoveryPath` doesn't result in allowing other data to be committed?

Reply:
Yes.